server/utils/messageCheck.js

```diff
@@ -0,0 +1,22 @@
+const validLinkTypes = /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})/gi;
+const linkRegex = new RegExp(validLinkTypes);
+
+const linkCheck = (parameter) => {
```

Comment:
This is not very efficient let's find a way to match the entire parameter against our regex instead of splitting, looping and checking.

We'll discuss more during our code review